openassessment/staffgrader/staff_grader_mixin.py

```diff
@@ -194,6 +194,39 @@ def list_staff_workflows(self, data, suffix=''):  # pylint: disable=unused-argum
                 log.exception("Failed to serialize workflow %d: %s", staff_workflow.id, str(e), exc_info=True)
         return result
 
+
+
+    @XBlock.json_handler
+    @require_course_staff("STUDENT_GRADE")
+    def list_assessments_grades(self, data, suffix=''):  # pylint: disable=unused-argument
+        """
+        List the assessments' grades based on the type (received or given) for a specific submission.
+
+        Args:
+            data (dict): Contains the necessary information to fetch the assessments.
+                        - 'item_id': The ID of the xblock/item.
+                        - 'submission_uuid': The UUID of the submission.
+                        - 'assessment_type': A string, either "received" or any other value 
+                                            to determine the type of assessments to retrieve.
+
+        Returns:
+            list[dict]: A list of dictionaries, each representing an assessment's data.
+
+        Note:
+            - If 'assessment_type' is "received", the function fetches assessments received 
+            for the given 'submission_uuid'.
+            - For any other value of 'assessment_type', the function fetches assessments 
+            given by the owner of the 'submission_uuid' for other submissions in the same item.
+        """
+        item_id = data['item_id']
+        subission_uuid = data['submission_uuid']
+
+        if data['assessment_type'] == "received":
+            return generate_received_assessment_data(subission_uuid)
+        else:
+            return generate_given_assessment_data(item_id, subission_uuid)
```

Comment:
is this `given` and `received` concept present elsewhere? can we elaborate on the meaning of each?

Reply:
I believe the default should be list all assessments for the submission uuid, so received.

Reply:
why is a single handler better than separating concerns? `list_assessments_received` or `list_assessments_given` are more explicit than `list_assessments...`, but I don't have other strong arguments

Reply:
In the MFE, it's just a table with a filter button; the data output has the same format. At the backend level, it's just a design decision that, compared to having two handlers, neither adds nor subtracts anything; it's pretty much the same.

Reply:
It's not the same. When using a single handler with a variable `assessment_type`, which defaults to given, you are making a decision there. When using two handlers, the client can actively decide which one to call without guessing the system's default. Don't you think?

Reply:
Yeah! that's true... but it does not exists a request without assessment_type (now called `assessment_filter`), it will always be `given` or `received`... 

![image](https://github.com/eduNEXT/edx-ora2/assets/12719909/72812f72-44a6-42e5-9ece-8777bdb70b22)


In fact, it is a required param in the `AssessmentFeedbackView` of edx-platform and declared in lms/djangoapps/ora_staff_grader/constants.py  so, list_assessments() would be never called without assessment_filter...  

jeje, but now tha you are saying this...  I think that in form, this will look even better  :smiley: 

```
        filter_value = data['assessment_filter']

        if filter_value == "received":
            return generate_received_assessment_data(subission_uuid)
        elif filter_value == "given":
            return generate_given_assessment_data(item_id, subission_uuid)
        else:
            raise ValueError("Invalid assessment_filter value")
```